pkgs/os-specific/darwin/cctools/llvm.nix

```
@@ -77,13 +87,16 @@ stdenv.mkDerivation {
     mkdir -p "$out/bin" "$man"
     ln -s ${lib.getDev cctools-port} "$dev"
 
+  '' + lib.optionalString useClangAssembler ''
     # Use the clang-integrated assembler instead of using `as` from cctools.
     makeWrapper "${lib.getBin llvmPackages.clang-unwrapped}/bin/clang" "$out/bin/${targetPrefix}as" \
       --add-flags "-x assembler -integrated-as -c"
 
+  '' + lib.optionalString useLLVMArchiver ''
     ln -s "${lib.getBin llvmPackages.bintools-unwrapped}/bin/llvm-ar" "$out/bin/${targetPrefix}ar"
     ${linkManPages llvmPackages.llvm-manpages "llvm-ar" "ar"}
```

My build finally failed, so I could `nix develop` and interactively look at what’s wrong. The issue isn’t an incompatibility. What’s happening is llvmPackages.bintools-unwrapped uses `targetPrefix`, which results in a dangling symlink in a cross build. The following patch fixes the issue for me.

```suggestion
    ln -s "${lib.getBin llvmPackages.bintools-unwrapped}/bin/${targetPrefix}llvm-ar" "$out/bin/${targetPrefix}ar"
    ${linkManPages llvmPackages.llvm-manpages "llvm-ar" "ar"}
```

---

The other changes can be dropped since they’re not necessary with this fix. The `enableClangAssembler` can also stay with #242202 since there’s a fix already for the GCC failure.

---

Done. Thank you!